From 3413445d2c9ab29b49844a946ca92d47dfd5cc4d Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Date: Wed, 18 Aug 2021 12:16:05 -0700
Subject: [PATCH 1906/1921] drivers: marvell: otx2-sdei-ghes: Add shutdown
 function

The SDEI GHES driver is missing shutdown function, is required
for graceful shutdown.

Change-Id: I920bfdf2fac7120ffc8738bfb19be56e88beb563
Signed-off-by: Vasyl Gomonovych <vgomonovych@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/59361
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-by: Linu Cherian <lcherian@marvell.com>
Tested-by: Linu Cherian <lcherian@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../marvell/octeontx2-ghes/otx2-ghes-bert.c   | 22 ++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/octeontx2-ghes/otx2-ghes-bert.c b/drivers/soc/marvell/octeontx2-ghes/otx2-ghes-bert.c
index a4716ee952c9..c7a69063e579 100644
--- a/drivers/soc/marvell/octeontx2-ghes/otx2-ghes-bert.c
+++ b/drivers/soc/marvell/octeontx2-ghes/otx2-ghes-bert.c
@@ -390,19 +390,33 @@ static int __init ghes_bert_probe(struct platform_device *pdev)
 	return ret;
 }
 
+static void ghes_bert_shutdown(struct platform_device *pdev)
+{
+	initerrmsg("%s: entry\n", __func__);
+}
+
+static int ghes_bert_remove(struct platform_device *pdev)
+{
+	initerrmsg("%s: entry\n", __func__);
+	return 0;
+}
+
+
 static const struct platform_device_id ghes_bert_pdev_match[] = {
 	{ .name = DRV_NAME, },
 	{},
 };
 MODULE_DEVICE_TABLE(platform, ghes_bert_pdev_match);
 
-static struct platform_driver ghes_bert_drv __initdata = {
+static struct platform_driver ghes_bert_drv = {
 	.driver = {
 		.name             = DRV_NAME,
 		.of_match_table   = bed_bert_of_match,
 		.acpi_match_table = ACPI_PTR(bed_bert_acpi_match),
 	},
 	.probe    = ghes_bert_probe,
+	.remove   = ghes_bert_remove,
+	.shutdown = ghes_bert_shutdown,
 	.id_table = ghes_bert_pdev_match,
 };
 
@@ -412,7 +426,13 @@ static int __init ghes_bert_init(void)
 	return 0;
 }
 
+static void __exit ghes_bert_exit(void)
+{
+	platform_driver_unregister(&ghes_bert_drv);
+}
+
 module_init(ghes_bert_init);
+module_exit(ghes_bert_exit);
 
 MODULE_DESCRIPTION("OcteonTX2 GHES BERT Module");
 MODULE_LICENSE("GPL v2");
-- 
2.31.1

